### ISSUE TITLE
error on `lsp-grammarly-stats`

### ISSUE THREAD

**Issue by vzaliva** [emacs-grammarly/lsp-grammarly#22]
error on `lsp-grammarly-get-document-state`:

`Error processing message (wrong-type-argument hash-table-p nil).`

`lsp-describe-session` shows:

```
[-] /home/lord/src/cheri-c-semantics
 `-[-] grammarly-ls:582471
    |-[-] Buffers
    |  `-[X] directions.org
    `-[-] Capabilities
       |-[X] codeActionProvider: t
       |-[X] hoverProvider: t
       `-[-] textDocumentSync:
          |-[X] change: 2
          `-[X] openClose: t
[X] /home/lord/tmp
```

Additionally (it might be related) my document does not seem to be checked. No warnings or errors are shown.

**Comment by vzaliva** [comment]
Stack trace:

```
Debugger entered--Lisp error: (wrong-type-argument hash-table-p nil)
  gethash("charsCount" nil nil)
  #f(compiled-function (state) #<bytecode 0x1592332412c5>)(#<hash-table equal 10/10 0x15923334f401>)
  #f(compiled-function (result) #<bytecode 0x15923362b419>)(#<hash-table equal 10/10 0x15923334f401>)
  #f(compiled-function (result) #<bytecode 0x15923362b435>)(#<hash-table equal 10/10 0x15923334f401>)
  lsp--parser-on-message(#<hash-table equal 3/3 0x15923334bd99> #s(lsp--workspace :ewoc nil :server-capabilities #<hash-table equal 3/3 0x1592336401ed> :registered-server-capabilities nil :root "/home/lord/src/cheri-c-semantics" :client #s(lsp--client :language-id nil :add-on? t :new-connection (:connect #f(compiled-function (filter sentinel name environment-fn) #<bytecode 0x15923324009d>) :test\? #f(compiled-function () #<bytecode 0x1592332400b1>)) :ignore-regexps nil :ignore-messages nil :notification-handlers #<hash-table equal 0/65 0x159233240395> :request-handlers #<hash-table equal 0/65 0x1592332403b5> :response-handlers #<hash-table eql 0/65 0x1592332407d9> :prefix-function nil :uri-handlers #<hash-table equal 0/65 0x159233240bd9> :action-handlers #<hash-table equal 0/65 0x1592332403d5> :major-modes nil :activation-fn #f(compiled-function (&rest _) #<bytecode 0x15923323da05>) :priority -1 :server-id grammarly-ls :multi-root nil :initialization-options nil :semantic-tokens-faces-overrides nil :custom-capabilities nil :library-folders-fn nil :before-file-open-fn nil :initialized-fn nil :remote? nil :completion-in-comments? nil :path->uri-fn nil :uri->path-fn nil :environment-fn nil :after-open-fn lsp-grammarly--init :async-request-handlers #<hash-table equal 5/65 0x1592332400bd> :download-server-fn #f(compiled-function (client callback error-callback update\?) #<bytecode 0x15923323da19>) :download-in-progress? nil :buffers nil :synchronize-sections nil) :host-root nil :proc #<process grammarly-ls> :cmd-proc #<process grammarly-ls> :buffers (#<buffer notes023-2021-11-22.org> #<buffer README.org>) :semantic-tokens-faces nil :semantic-tokens-modifier-faces nil :extra-client-capabilities nil :status initialized :metadata #<hash-table equal 0/65 0x159233537b11> :watches #<hash-table equal 0/65 0x159233537b31> :workspace-folders nil :last-id 0 :status-string nil :shutdown-action nil :diagnostics #<hash-table equal 0/65 0x159233537b51> :work-done-tokens #<hash-table equal 0/65 0x1592334f7705>))
  #f(compiled-function (msg) #<bytecode 0x15923334f529>)(#<hash-table equal 3/3 0x15923334bd99>)
  mapc(#f(compiled-function (msg) #<bytecode 0x15923334f529>) (#<hash-table equal 3/3 0x15923334bd99>))
  #f(compiled-function (proc input) #<bytecode 0x159233669461>)(#<process grammarly-ls> "Content-Length: 343\15\n\15\n{\"jsonrpc\":\"2.0\",\"id\":147,\"result\":{\"uri\":\"file:///home/lord/src/cheri-c-semantics/notes/notes023-2021-11-22.org\",\"score\":-1,\"status\":\"CHECKING\",\"scores\":{},\"emotions\":[],\"textInfo\":null,\"totalAlertsCount\":0,\"additionalFixableAlertsCount\":0,\"premiumAlertsCount\":0,\"user\":{\"username\":\"lord@crocodile.org\",\"isAnonymous\":false,\"isPremium\":true}}}")
```

**Comment by vzaliva** [comment]
So, the problem seems to be that `textInfo` is `null` in the server response.

**Comment by jcs090218** [comment]
Can you paste the content here? I could not get `text-info` to become `null`. Thanks!

**Comment by vzaliva** [comment]
File test.org:

```
* Test
  Some Engrish texto.

  
```

**Comment by jcs090218** [comment]
I have applied a fix in 845dd407d3dc4e0cd419390793a9d3f3f2f80704.

Didn't know that `textInfo` can be `null` in the first place. 😕
